halo2_proofs/CHANGELOG.md

```diff
@@ -10,6 +10,18 @@ and this project adheres to Rust's notion of
 - `halo2_proofs::circuit::Value`, a more usable and type-safe replacement for
   `Option<V>` in circuit synthesis.
 - `impl Mul<F: Field> for &Assigned<F>`
+- `halo2_proofs::plonk`:
```

Comment:
Now that we've released `halo2_proofs 0.2.0`, this PR needs to be rebased (because this section of the changelog is now part of `0.2.0`. Additionally, the commits should be squashed together so that the hard-coded proof binary is unaltered by the branch (rather than having multiple changes across several commits that together are a no-op but cost commit size).

Reply:
Make sense, I have rebased and squashed all commits into 6 ones with suggestions and fixes above (and I resolved some minor comments, only leave those I'm not sure how to resolve for further discussion), hope it look cleaner now.